### ISSUE TITLE
How to make this work with SASS extension

### ISSUE THREAD

**Issue by olivermuc** [y0ssar1an/CSS3#95]
It looks like I can't have the cake and eat it, too.

The problem: CSS3 extension requires the "file type" to be set to CSS3 to take effect.
The same goes for the SASS extension.
So it appears that both extensions are mutually exclusive, as I can't configure any to also be active when the other file type is being loaded.

Any ideas on how to enable both CSS3 autocompletion + SASS autocompletion for a *.scss file?

Thanks!
Oliver

**Comment by ellipticaldoor** [comment]
+1

**Comment by roblav96** [comment]
Yes please!

**Comment by y0ssar1an** [comment]
First, I'm sorry for the late reply. I've been in a cave studying for job interviews. Second, I'm sorry but SASS support is not possible. The syntax definition is 10k lines that are highly specific to CSS syntax. SCSS syntax diverges too much from CSS.

As an alternative, I can only suggest [cssnext](http://cssnext.io), which is fully supported. It has most of the SASS features you want, with a syntax that arguably _is_ CSS (because it's based on W3C draft specs).

**Comment by olivermuc** [comment]
Thanks for the reply @y0ssar1an 
Not sure if it changes anything, but I wasn‘t looking for Sass syntax support from your extention.
Merely a configuration option that allows your extension as-is to play nicely with eg an existing Sass extension.
Eg, if I could tell your extension to also be active when other fily types are being loaded, not just *.css than that would already help a ton.

**Comment by y0ssar1an** [comment]
Sorry I misunderstood your issue earlier. The two major services provided by CSS3 are 1) syntax highlighting and 2) completions. I thought you were asking for 1, but really you were asking for 2. If I understand you correctly, you're asking if it's possible to use CSS3's completions when you're editing a SASS file.

Sadly, I have to disappoint you again. I installed [the Sass package](https://github.com/P233/Syntax-highlighting-for-Sass) and looked at its syntax definition to see if this was possible. Completions work hand-in-hand with the syntax definition. The syntax definition applies labels (aka "scopes") to the text. For example, 
```css
h1 {
    colo|r: red;  
}
```
The cursor is on the `color` property above. These are the labels applied by the CSS3 syntax definition.
```
source.css
meta.rule.css
meta.declaration-list.css
support.type.property-name.css
```
The completions system looks for those labels under the cursor. If the cursor is on text with the right label, the completions menu pops up.

The problem is that the Sass syntax definition applies its own labels to the text, and my completions were written specifically for the labels in [`CSS3.sublime-syntax`](https://github.com/y0ssar1an/CSS3/blob/master/CSS3.sublime-syntax). It's possible to write completions that would work for both CSS code and SCSS code. The Sass authors left the door open for that by using labels like [`support.type.property-name.css.scss`](https://github.com/P233/Syntax-highlighting-for-Sass/blob/master/Syntaxes/SCSS.sublime-syntax#L165). Unfortunately, when I wrote the completions I was just trying to get them working with CSS3 labels. I wanted the completions to be highly specific to what the user is typing. That meant applying lots of custom labels. The Sass package doesn't apply those labels, so the CSS3 completions wouldn't be triggered even if I modified them to permit `source.scss`.

I think the only way this could ever happen is to write a new syntax definition and completions from scratch, with the explicit goal of supporting both CSS and SCSS. That's a huge undertaking and I have no plans to do that, sorry!
